src/index.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  BN,
+  keccak,
+  keccak256,
+  rlphash,
+  zeros,
+  bufferToInt,
+  TWO_POW256
+} from 'ethereumjs-util'
+import {
+  params,
+  fnv,
+  fnvBuffer,
+  bufReverse,
+  getEpoc,
+  getCacheSize,
+  getFullSize,
+  getSeed
+} from './util'
+import type { LevelUp } from 'levelup'
+const async = require('async')
+const xor = require('buffer-xor')
+
+// ethereumjs-block v3 will have types
+type Block = any
+type BlockHeader = any
+
+export default class Ethash {
+  dbOpts: Object
+  cacheDB?: LevelUp
+  cache: Buffer[]
+  epoc?: number
+  fullSize?: number
+  cacheSize?: number
+  seed?: Buffer
+
+  constructor(cacheDB?: LevelUp) {
+    this.dbOpts = {
+      valueEncoding: 'json'
+    }
+    this.cacheDB = cacheDB
+    this.cache = []
```

Comment:
Whew, a lot better initialization, this was really confusing to have this set to `false` in the JS version.